packages/core/upload/admin/src/components/AssetDialog/BrowseStep/BrowseStep.tsx

```diff
@@ -179,7 +181,7 @@ export const BrowseStep = ({
     (asset) => selectedAssets.findIndex((currAsset) => currAsset.id === asset.id) !== -1
   );
   const isSearching = !!queryObject?._q;
-  const isFiltering = queryObject?.filters?.$and?.length! > 0;
+  const isFiltering = queryObject?.filters?.$and?.length && queryObject.filters.$and.length > 0;
```

Comment:
would this be the same?
```suggestion
  const isFiltering = queryObject?.filters?.$and?.length!!
```

Reply:
Using the Boolean to cast the value gave me the Type error in the second part of the condition, 
<img width="953" alt="Screenshot 2024-11-04 alle 10 08 29" src="https://github.com/user-attachments/assets/23ec35ff-b789-4291-8277-b10edd324e9b">
otherwise the solution suggested by @Marc-Roig works

Reply:
Sorry I think this is what I had in mind. 
```suggestion
  const isFiltering = Boolean(queryObject?.filters?.$and?.length > 0);
```

But Marc's solution is fine too, my comment was just a code style suggestion so whichever you prefer is fine by me.